Website/src/rendering/src/components/FullPageSearch/FullPageSearch.tsx

```diff
@@ -40,136 +25,28 @@ const FullPageSearch = ({
   onPageNumberChange,
   onSortChange,
 }: FullPageSearchResultsProps): JSX.Element => {
-  const isCategoryProductListingPage = rfkId === 'rfkid_10';
-
-  const [toggle, setToggle] = useState(false);
-
-  const setKeyphrase: (keyphrase: string) => void = debounce(
-    (keyphrase) =>
-      dispatch({ type: SearchResultsActions.KEYPHRASE_CHANGED, payload: { keyphrase } }),
-    500,
-    false
-  );
-
-  useEffect(() => {
-    const urlSearchParams = new URLSearchParams(window.location.search);
-    const searchQuery = urlSearchParams.get('q');
-    const keyphraseToUse = keyphrase ?? searchQuery;
-    if (keyphraseToUse) {
-      setKeyphrase(keyphraseToUse);
-    }
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, []);
-
-  if (error) {
-    return <div>Response error</div>;
-  }
-
-  const handleFacetClick = (payload: SearchResultsFacetClickedChangedActionPayload) => {
-    onFacetClick(payload);
-  };
-
-  const handleFacetClear = () => {
-    onClearFilters();
-  };
-
-  const handlePageNumberChange = (pageNumber: string) => {
-    const pageNo: SearchResultsPageNumberChangedActionPayload = {
-      rfkId,
-      page: Number(pageNumber),
-    };
-    onPageNumberChange(pageNo);
-  };
-
-  const handleSortChange = (payload: SearchResultsSortChangedActionPayload) => {
-    onSortChange(payload);
-  };
-
-  const handleToggleClick = () => {
-    const isVisible = !toggle;
-    setToggle(isVisible);
-    document.body.classList.toggle('shop-facet-panel-open', isVisible);
-  };
-
-  const handleSearchInputChange = (e: ChangeEvent<HTMLInputElement>) => {
-    setKeyphrase(e.target.value || '');
-  };
-
-  const numberOfResults = !loading && totalPages > 0 && (
-    <div className="items-num">{totalItems} items</div>
-  );
-
-  const noResultsMessage = totalItems === 0 && 'No results found';
-
-  const sortFacetProps = {
-    sortChoices,
-    sortType,
-    sortDirection,
-    onSortChange: handleSortChange,
-  };
-
-  // TODO: Extract this whole component except this line that gets the URL path into a FullPageSearchContent.tsx component that will accept an extra prop for the urlPath. Then rename the FullPageSearch.stories.tsx to FullPageSearchContent.stories.tsx and test the sub component. Create test cases when the urlPath is for a non-existing category, an existing category, etc.
-  const category = getCategoryByUrlPath(window.location.pathname);
```

Comment:
I feel like the getCategoryByUrlPath should stay in this component and the FullPageSearchContent would then get the category without having to worry where it's coming from. This also applies to the `keyphraseToUse` above (line 57).

Reply:
And we should directly pass the category to FullPageSearchContent instead of the urlPath you mean?

Reply:
Just an idea but yes, the category and the keyphraseToUse if there is one. Becase currently the FullPageSearch component has no reason to exist - it's only passing `window.location.pathname` as a prop, which is unnecessary since the FullPageSearchContent can access that by itself.